src/main/java/homework_4/singleton/Singleton.java

```diff
@@ -0,0 +1,13 @@
+package homework_4.singleton;
+
+public final class Singleton {
+    private Singleton() {}
+
+    private static class Holder {
+        static final Singleton INSTANCE = new Singleton();
```

Comment:
Opt: why not just have static variable?

Reply:
Lazy init.

Reply:
The JVM defers initializing the InstanceHolder class until it is actually used, and because the Singleton is initialized with a static initializer, no additional synchronization is needed. The first call to getInstance by any thread causes Holder to be loaded and initialized, at which time the initialization of the Singleton happens through the static initializer.

Reply:
Agree, done